hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/GrpcBlockItemWriter.java

```diff
@@ -0,0 +1,214 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.blocks.impl;
+
+import static io.grpc.Status.fromThrowable;
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hedera.hapi.block.protoc.BlockStreamServiceGrpc;
+import com.hedera.hapi.block.protoc.PublishStreamRequest;
+import com.hedera.hapi.block.protoc.PublishStreamResponse;
+import com.hedera.hapi.block.protoc.PublishStreamResponse.Acknowledgement;
+import com.hedera.hapi.block.stream.protoc.BlockItem;
+import com.hedera.node.app.blocks.BlockItemWriter;
+import com.hedera.node.config.data.BlockStreamConfig;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Implements the bidirectional streaming RPC for the publishBlockStream rpc in BlockStreamService
+ * See <a href="https://grpc.io/docs/languages/java/basics/">gRPC Basics</a>
+ */
+public class GrpcBlockItemWriter implements BlockItemWriter {
+
+    private static final Logger logger = LogManager.getLogger(GrpcBlockItemWriter.class);
+    private static final String INVALID_MESSAGE = "Invalid protocol buffer converting %s from PBJ to protoc for %s";
+
+    private final BlockStreamServiceGrpc.BlockStreamServiceStub asyncStub;
+    private final ManagedChannel channel;
+    private StreamObserver<PublishStreamRequest> requestObserver;
+
+    private long blockNumber;
+
+    /** The state of this writer */
+    private State state = State.UNINITIALIZED;
+
+    /**
+     * The current state of the gRPC writer.
+     */
+    public enum State {
+        /**
+         * The gRPC client still not initialized
+         */
+        UNINITIALIZED,
+        /**
+         * The gRPC client is currently open
+         */
+        OPEN,
+        /**
+         * The gRPC client is already closed
+         */
+        CLOSED
+    }
+
+    /**
+     * @param blockStreamConfig the block stream configuration
+     */
+    public GrpcBlockItemWriter(@NonNull final BlockStreamConfig blockStreamConfig) {
+        requireNonNull(blockStreamConfig, "The supplied argument 'blockStreamConfig' cannot be null!");
+
+        channel = ManagedChannelBuilder.forAddress(blockStreamConfig.address(), blockStreamConfig.port())
+                .usePlaintext()
```

Comment:
I'm assuming communication via plaintext is temporary?

Reply:
@mhess-swl 
I was originating from the communication between the Java SDK and services in [BaseNode.java](https://github.com/hashgraph/hedera-sdk-java/blob/main/sdk/src/main/java/com/hedera/hashgraph/sdk/BaseNode.java#L281) where we provide a plaintext communication, if we are not using a TLS communication.